File: stdlib/cpp/MathTools.cpp
#pragma once

#include <math.h>
#include <any>

static auto arcsine(auto y)
{
    any x = (any)y;
    if (x.type() == typeid(double))
        return any_cast<double>(asin(*any_cast<double>(&x)));
}

static auto arccos(auto y)
{
    any x = (any)y;
    if (x.type() == typeid(double))
        return any_cast<double>(acos(*any_cast<double>(&x)));
}

static auto cosine(auto y)
{
    any x = (any)y;
    if (x.type() == typeid(double))
        return any_cast<double>(cos(*any_cast<double>(&x)));
}

static auto sine(auto y)
{
    any x = (any)y;
    if (x.type() == typeid(double))
        return any_cast<double>(sin(*any_cast<double>(&x)));
}

static auto floorValue(auto y)
{
    any x = (any)y;
    if (x.type() == typeid(double))
        return any_cast<double>(floor(*any_cast<double>(&x)));
}

File: stdlib/go/MathTools.go
package main

import (
	"math"
)

func arcsine(x2 dynamic) dynamic {
	return math.Asin(x2.(float64))
}

func arccos(x2 dynamic) dynamic {
	return math.Acos(x2.(float64))
}

func cosine(x3 dynamic) dynamic {
	return math.Cos(x3.(float64))
}

func sine(x5 dynamic) dynamic {
	return math.Sin(x5.(float64))
}

func floorValue(x6 dynamic) dynamic {
	return math.Floor(x6.(float64))
}
